hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -84,56 +85,49 @@ public ResponseCodeEnum wipe(final AccountID aId, final TokenID tId, final long
 	@Override
 	public CreationResult<List<Long>> mint(final TokenMintTransactionBody txBody, final RichInstant creationTime) {
 		var provisionalUniqueTokens = new ArrayList<Pair<MerkleUniqueTokenId, MerkleUniqueToken>>();
-
 		var tokenId = txBody.getToken();
 		var provisionalSanityCheck = tokenSanityCheck(tokenId, merkleToken -> {
-
 			if (!merkleToken.hasSupplyKey()) {
 				return TOKEN_HAS_NO_SUPPLY_KEY;
 			}
-
 			final var eId = EntityId.fromGrpcTokenId(tokenId);
 			final var owner = merkleToken.treasury();
 			final var metadataList = txBody.getMetadataList();
 			long serialNum = merkleToken.getCurrentSerialNum();
-
 			for (ByteString el : metadataList) {
 				serialNum++;
 				String metaAsStr = el.toStringUtf8();
 				final var nftId = new MerkleUniqueTokenId(eId, serialNum);
 				final var nft = new MerkleUniqueToken(owner, metaAsStr, creationTime);
 				provisionalUniqueTokens.add(Pair.of(nftId, nft));
 			}
-
 			if (!checkProvisional(provisionalUniqueTokens)) {
 				return INVALID_TRANSACTION_BODY;
 			}
-
 			var adjustmentResult = tryAdjustment(merkleToken.treasury().toGrpcAccountId(), tokenId, provisionalUniqueTokens.size());
-
 			if (!adjustmentResult.equals(OK)) {
 				return adjustmentResult;
 			}
-
-				return OK;
+			return OK;
 		});
-
 		if (!provisionalSanityCheck.equals(OK)) {
 			return CreationResult.failure(provisionalSanityCheck);
 		}
+		// Commit logic
+		var token = super.getTokens().get().getForModify(tokenId);
 		var lastMintedSerialNumbers = new ArrayList<Long>();
-		var token = get(tokenId);
-		provisionalUniqueTokens.forEach(pair -> {
+		for (Pair<MerkleUniqueTokenId, MerkleUniqueToken> pair : provisionalUniqueTokens) {
 			var nft = pair.getValue();
 			var nftId = pair.getKey();
-			lastMintedSerialNumbers.add(token.incrementSerialNum());
+			var serialNum = token.incrementSerialNum();
```

Comment:
Wouldn't it be better if we modify the `serialNumber` after we iterated through all of the pairs. That we we will do only only update of the property instead of `n`